core/deployment/src/main/java/io/quarkus/runner/bootstrap/StartupActionImpl.java

```diff
@@ -0,0 +1,170 @@
+                    try {
+                        try {
+                            closeTask.close();
+                        } finally {
+                            runtimeClassLoader.close();
+                        }
+                    } finally {
+                        if (curatedApplication.getQuarkusBootstrap().getMode() == QuarkusBootstrap.Mode.TEST) {
+                            //for tests we just always shut down the curated application, as it is only used once
+                            //dev mode might be about to restart, so we leave it
+                            curatedApplication.close();
+                        }
+                    }
+                }
+            }, runtimeClassLoader);
+        } catch (InvocationTargetException e) {
+            if (e.getCause() instanceof Exception) {
+                throw (Exception) e.getCause();
+            }
+            throw new RuntimeException("Failed to start Quarkus", e.getCause());
+        } finally {
+            Thread.currentThread().setContextClassLoader(old);
+        }
+
+    }
+
+    private Map<String, List<BiFunction<String, ClassVisitor, ClassVisitor>>> extractTransformers() {
+        Map<String, List<BiFunction<String, ClassVisitor, ClassVisitor>>> bytecodeTransformers = new HashMap<>();
+        List<BytecodeTransformerBuildItem> transformers = buildResult.consumeMulti(BytecodeTransformerBuildItem.class);
+        for (BytecodeTransformerBuildItem i : transformers) {
+            List<BiFunction<String, ClassVisitor, ClassVisitor>> list = bytecodeTransformers.get(i.getClassToTransform());
+            if (list == null) {
+                bytecodeTransformers.put(i.getClassToTransform(), list = new ArrayList<>());
+            }
+            list.add(i.getVisitorFunction());
+        }
+        return bytecodeTransformers;
+    }
+
+    private Map<String, byte[]> extractGeneratedResources(boolean applicationClasses) {
+        Map<String, byte[]> data = new HashMap<>();
+        for (GeneratedClassBuildItem i : buildResult.consumeMulti(GeneratedClassBuildItem.class)) {
+            if (i.isApplicationClass() == applicationClasses) {
+                data.put(i.getName().replace(".", "/") + ".class", i.getClassData());
+                if (DEBUG_CLASSES_DIR != null) {
+                    try {
+                        File debugPath = new File(DEBUG_CLASSES_DIR);
+                        if (!debugPath.exists()) {
+                            debugPath.mkdir();
+                        }
+                        File classFile = new File(debugPath, i.getName() + ".class");
+                        classFile.getParentFile().mkdirs();
+                        try (FileOutputStream classWriter = new FileOutputStream(classFile)) {
+                            classWriter.write(i.getClassData());
+                        }
+                        log.infof("Wrote %s", classFile.getAbsolutePath());
+                    } catch (Throwable t) {
```

Comment:
Does it really have to be a Throwable? Swallowing throwables in a loop may go pretty bad.